src/pluggy/_tracing.py

```diff
@@ -3,17 +3,39 @@
 """
 from .callers import _Result
 
+if False:  # TYPE_CHECKING
+    from typing import Any
+    from typing import Callable
+    from typing import Dict
+    from typing import List
+    from typing import Optional
+    from typing import Sequence
+    from typing import Tuple
+    from typing import Union
+
+    from .hooks import HookImpl
+    from .hooks import _HookCaller
+    from .manager import PluginManager
+
+    _Writer = Callable[[str], None]
+    _Processor = Callable[[Tuple[str, ...], Sequence[object]], None]
```

Comment:
As you probably know, `object` and `Any` both accept any type, but they are opposites in terms of what you can do with them: with `object`, you can't do much of anything (just `__str__` and stuff) without narrowing it first with `isinstance`, while `Any` permits anything.

The rule I usually follow is: for *inputs* that can be anything, I use `object` - this puts more restrictions on me and provides more assurances to the caller.  For outputs, I use `Any`, since I don't want to force the caller to perform `isinstance` checks.

(Note: when the use provides a callback to be called by the library, the roles reverse, that is the arguments become `Any` and the return type becomes `object`).

I probably got it wrong in a few places so I'll go over it.